scripts/tests/api_compare/docker-compose.yml

```diff
@@ -2,70 +2,185 @@
+      dockerfile: ${FOREST_DOCKERFILE_OVERRIDE:-Dockerfile}
+    volumes:
+      - node-data:/data
+      - filecoin-proofs:${FIL_PROOFS_PARAMETER_CACHE}
+    networks:
+      - api-tests
+    environment:
+      - FIL_PROOFS_PARAMETER_CACHE=${FIL_PROOFS_PARAMETER_CACHE}
+      - FULLNODE_API_INFO=/dns/forest/tcp/${FOREST_RPC_PORT}/http
+    entrypoint: [ "/bin/bash", "-c" ]
+    command:
+      - |
+        set -euxo pipefail
+        # Setup data_dir to avoid downloading snapshot on every local run
+        echo "[client]" > config.toml
+        echo "data_dir = \"/data/forest\"" >> config.toml
+        forest --config config.toml --chain ${CHAIN} --encrypt-keystore false --no-gc --height=-900 --auto-download-snapshot --rpc-address 0.0.0.0:${FOREST_RPC_PORT}
+    healthcheck:
+      test: [ "CMD", "forest-cli", "chain", "head" ]
+      interval: 15s
+      timeout: 3s
+      retries: 3
+  snapshot-export:
+    depends_on:
+      forest:
+        condition: service_healthy
+    build:
+      context: ../../../.
+      dockerfile: ${FOREST_DOCKERFILE_OVERRIDE:-Dockerfile}
+    networks:
+      - api-tests
     volumes:
       - node-data:/data
       - filecoin-proofs:/proofs
     env_file:
       - .env
-    entrypoint: ["/bin/bash", "-c" ]
+    environment:
+      - FULLNODE_API_INFO=/dns/forest/tcp/${FOREST_RPC_PORT}/http
+    entrypoint: [ "/bin/bash", "-c" ]
     command:
       - |
         set -euxo pipefail
-        # if there are some files in the data directory, then we don't need to fetch the snapshot
-        if [ "$$(ls -A /data/snapshot/)" ]; then
-          echo "Snapshot already fetched"
-        else
-          mkdir -p /data/snapshot
-          forest-tool snapshot fetch  --chain ${CHAIN} --directory /data/snapshot/
-        fi
-        # copy proofs to the shared volume, if they are not already there
-        if [ "$$(ls -A /proofs/)" ]; then
-          cp -r ${FIL_PROOFS_PARAMETER_CACHE}/* /proofs/
-        fi
-  forest:
+        forest-cli sync wait
+        forest-cli snapshot export -o "$SNAPSHOT_PATH" -d=900 --include-message-receipts
+  api-serve:
     depends_on:
-      init:
+      snapshot-export:
         condition: service_completed_successfully
-    image: ${FOREST_IMAGE}
+    build:
+      context: ../../../.
+      dockerfile: ${FOREST_DOCKERFILE_OVERRIDE:-Dockerfile}
     volumes:
       - node-data:/data
       - filecoin-proofs:${FIL_PROOFS_PARAMETER_CACHE}
     networks:
       - api-tests
     environment:
       - FIL_PROOFS_PARAMETER_CACHE=${FIL_PROOFS_PARAMETER_CACHE}
-    entrypoint: ["/bin/bash", "-c" ]
+    entrypoint: [ "/bin/bash", "-c" ]
     command:
       - |
         set -euxo pipefail
-        forest --import-snapshot /data/snapshot/* --encrypt-keystore false --chain ${CHAIN} --rpc-address 0.0.0.0:${FOREST_RPC_PORT}
+        forest-tool api serve "$SNAPSHOT_PATH" \
+          --chain calibnet --port ${FOREST_OFFLINE_RPC_PORT}
   lotus:
     depends_on:
-      init:
+      snapshot-export:
```

Comment:
Would it make sense to make certain steps optional? For example, I'm testing a particular endpoint unrelated to snapshot exports and would like to exclusively run the offline tests. Setting a few variables instead of commenting out chunks of this file would be helpful. What do you think?

Reply:
I'm not sure if this is feasible as lotus needs the be bootstrapped with this exported snapshot that contains message receipts to test against. How about tracking this feature separately?